apps/web/app/(basenames)/api/basenames/metadata/[tokenId]/route.ts

```diff
@@ -1,37 +1,42 @@
-import { Basename } from '@coinbase/onchainkit/identity';
-import { premintMapping } from 'apps/web/pages/api/basenames/metadata/premintsMapping';
-import L2Resolver from 'apps/web/src/abis/L2Resolver';
-import { USERNAME_L2_RESOLVER_ADDRESSES } from 'apps/web/src/addresses/usernames';
-import { isDevelopment } from 'apps/web/src/constants';
-import { getBasenamePublicClient } from 'apps/web/src/hooks/useBasenameChain';
-import { logger } from 'apps/web/src/utils/logger';
+import { NextRequest, NextResponse } from 'next/server';
+import { base } from 'viem/chains';
 import {
   formatBaseEthDomain,
   getBasenameNameExpires,
   USERNAME_DOMAINS,
 } from 'apps/web/src/utils/usernames';
-import { NextResponse } from 'next/server';
 import { encodePacked, keccak256, namehash, toHex } from 'viem';
-import { base } from 'viem/chains';
+import { getBasenamePublicClient } from 'apps/web/src/hooks/useBasenameChain';
+import { USERNAME_L2_RESOLVER_ADDRESSES } from 'apps/web/src/addresses/usernames';
+import L2Resolver from 'apps/web/src/abis/L2Resolver';
+import { Basename } from '@coinbase/onchainkit/identity';
+import { logger } from 'apps/web/src/utils/logger';
+import { premintMapping } from 'apps/web/app/(basenames)/api/basenames/metadata/premintsMapping';
+import { getChain } from 'apps/web/src/utils/basenames/getChain';
+import { getDomain } from 'apps/web/src/utils/basenames/getDomain';
 
-export const config = {
-  runtime: 'edge',
-};
+export const runtime = 'edge';
 
-export default async function GET(request: Request) {
-  const url = new URL(request.url);
+export async function GET(request: NextRequest) {
+  const domainName = getDomain(request);
 
-  const domainName = isDevelopment ? `${url.protocol}//${url.host}` : 'https://www.base.org';
-  let tokenId = url.searchParams.get('tokenId');
-  if (tokenId?.endsWith('.json')) tokenId = tokenId.slice(0, -5);
-  const chainIdFromParams = url.searchParams.get('chainId');
-  const chainId = chainIdFromParams ? Number(chainIdFromParams) : base.id;
-  const baseDomainName = USERNAME_DOMAINS[chainId];
+  let tokenId = request.nextUrl.searchParams.get('tokenId');
+  if (tokenId?.endsWith('.json')) {
+    tokenId = tokenId.slice(0, -5);
+  }
```

Comment:
suggestion(non-blocking):

```ts
const tokenId = request.nextUrl.searchParams
      .get('tokenId')
      ?.replace(/\.json$/, '');
      // Or just ".json" if instead of regex,
      //but I wanted to keep the functionality the same
```

Reply:
nice, updated